tests/ui_tools/test_boxes.py

```diff
@@ -67,6 +67,52 @@ def test_not_calling_typing_method_without_recipients(self, mocker, write_box):
 
         assert not write_box.model.send_typing_status_by_user_ids.called
 
+    @pytest.mark.parametrize(
+        "text, state, required_typeahead, is_valid_stream",
+        [
+            ("#**Stream 1>T", 0, "#**Stream 1>Topic 1**", True),
+            ("#**Stream 1>T", 1, "#**Stream 1>This is a topic**", True),
+            ("#**Stream 1>T", 2, None, True),
+            ("#**Stream 1>T", -1, "#**Stream 1>This is a topic**", True),
+            ("#**Stream 1>T", -2, "#**Stream 1>Topic 1**", True),
+            ("#**Stream 1>T", -3, None, True),
+            ("#**Stream 1>To", 0, "#**Stream 1>Topic 1**", True),
+            ("#**Stream 1>H", 0, "#**Stream 1>Hello there!**", True),
+            ("#**Stream 1>Hello ", 0, "#**Stream 1>Hello there!**", True),
+            ("#**Stream 1>", 0, "#**Stream 1>Topic 1**", True),
+            ("#**Stream 1>", 1, "#**Stream 1>This is a topic**", True),
+            ("#**Stream 1>", -1, "#**Stream 1>Hello there!**", True),
+            ("#**Stream 1>", -2, "#**Stream 1>This is a topic**", True),
+            # Invalid stream
+            ("#**invalid stream>", 0, None, False),
+            # Invalid prefix format
+            ("#*Stream 1>", 0, None, True),
+            ("#Stream 1>", 0, None, True),
```

Comment:
This seems like it should complete?

Reply:
Well, for a user, autocompleting a stream name would be handier than fully writing it, I believe. Also, this doesn't cope up with the web app's way, I think.